test/Microsoft.ApplicationInsights.AspNetCore.Tests/JavaScript/ApplicationInsightsJavaScriptTest.cs

```diff
@@ -10,27 +15,93 @@ public static class ApplicationInsightsJavaScriptTest
         public static void SnippetWillBeEmptyWhenInstrumentationKeyIsNotDefined()
         {
             var telemetryConfigurationWithNullKey = new TelemetryConfiguration();
-            var snippet = new JavaScriptSnippet(telemetryConfigurationWithNullKey);
-            Assert.Equal(string.Empty, snippet.FullScript.ToString());
+            var snippet = new JavaScriptSnippet(telemetryConfigurationWithNullKey, GetOptions(false), null);
+            Assert.Equal(string.Empty, snippet.FullScript);
         }
 
         [Fact]
         public static void SnippetWillBeEmptyWhenInstrumentationKeyIsEmpty()
         {
-            var telemetryConfigurationWithEmptyKey = new TelemetryConfiguration { InstrumentationKey = string.Empty };
-            var snippet = new JavaScriptSnippet(telemetryConfigurationWithEmptyKey);
-            Assert.Equal(string.Empty, snippet.FullScript.ToString());
+            var telemetryConfigurationWithEmptyKey = new TelemetryConfiguration {InstrumentationKey = string.Empty};
+            var snippet = new JavaScriptSnippet(telemetryConfigurationWithEmptyKey, GetOptions(false), null);
+            Assert.Equal(string.Empty, snippet.FullScript);
         }
 
+        [Fact]
+        public static void SnippetWillBeEmptyWhenTelemetryDisabled()
+        {
+            var telemetryConfigurationWithEmptyKey = new TelemetryConfiguration
+            {
+                InstrumentationKey = "NonEmpty",
+                DisableTelemetry = true
+            };
+            var snippet = new JavaScriptSnippet(telemetryConfigurationWithEmptyKey, GetOptions(false), null);
+            Assert.Equal(string.Empty, snippet.FullScript);
+        }
 
         [Fact]
         public static void SnippetWillIncludeInstrumentationKeyAsSubstring()
         {
             string unittestkey = "unittestkey";
+            var telemetryConfiguration = new TelemetryConfiguration {InstrumentationKey = unittestkey};
+            var snippet = new JavaScriptSnippet(telemetryConfiguration, GetOptions(false), null);
+            Assert.Contains("instrumentationKey: '" + unittestkey + "'", snippet.FullScript);
+        }
+
+        [Fact]
+        public static void SnippetWillIncludeAuthUserNameIfEnabledAndAuthenticated()
+        {
+            string unittestkey = "unittestkey";
             var telemetryConfiguration = new TelemetryConfiguration { InstrumentationKey = unittestkey };
-            var snippet = new JavaScriptSnippet(telemetryConfiguration);
-            Assert.Contains("instrumentationKey: '" + unittestkey + "'", snippet.FullScript.ToString());
+            var snippet = new JavaScriptSnippet(telemetryConfiguration, GetOptions(true), GetHttpContextAccessor("username", true));
+            Assert.Contains("setAuthenticatedUserContext(\"username\")", snippet.FullScript);
+        }
+
+        [Fact]
+        public static void SnippetWillNotIncludeAuthUserNameIfEnabledAndAuthenticated()
```

Comment:
Maybe I'm misunderstanding what is happening, does the second parameter to GetHttpContextAccessor specify if a value is returned?  Could you add some doc comments to these test methods to explain exactlky what and how they are testing?  Thank-you.